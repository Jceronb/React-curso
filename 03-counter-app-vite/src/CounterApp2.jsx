
import PropTypes from 'prop-types';

export const CounterApp2 = ({values} ) => {
  return (
    <>
      <h1>Mi segundo CounterApp</h1>
      <h2>{values}</h2>
    </>
  )
}

CounterApp2.propTypes={
    values:PropTypes.number.isRequired,
}



