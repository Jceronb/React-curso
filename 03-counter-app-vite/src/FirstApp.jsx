import PropTypes from 'prop-types';



//const numeros = [1,2,3,4,5,6,7,8,9];
 
//const resultado =(a, b)=>{
    //return a * b;
//}

export const FirstApp = ({title,subTitle, name})=> {
  
       //console.log (props);
    return (
        <> 
        <h1>{title}</h1>
        <p>{subTitle} </p>
        <p>{name}</p>
        </>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string,
}

// Se puede definir los defaultProps

FirstApp.defaultProps = {
    name: 'Jesús Cerón',
    subTitle: 'No hay sub título',
    title: 'No hay título',

}


