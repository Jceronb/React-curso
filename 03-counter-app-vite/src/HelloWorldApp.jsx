
//Con una funcion tradicional
 export function App(){
     //document.getElement....
     return (<h1>Hola mundo !!!</h1> );
 }

 export const HelloWorldApp = () => {
    return(<h2>Hello World App</h2> )
 }

 //podemos crear estilos de CSS en una variable
let div = {
    width:150,
    color: 'red',
    backgroundColor:'black',
    fontSize: 25
}

export const App2 = ()=>(<div style={div}> Cómo estás? </div> );

