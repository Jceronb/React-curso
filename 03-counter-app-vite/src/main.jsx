import React from 'react';
import ReactDOM from 'react-dom/client';
//import { App, HelloWorldApp } from "./HelloWorldApp";
//import { App2 } from "./HelloWorldApp"
//import { FirstApp } from './FirstApp';

import './styles.css';
//import { CounterApp2 } from './CounterApp2';
import { CounterApp} from './CounterApp';


// Con una funcion tradicional
// function App(){
//     //document.getElement....
//     return (<h1>Hola mundo !!</h1> );
// }


// Con funcion flecha
//const App = ()=>(<h1>Hola mundo !!</h1> );
//const App2 = ()=>(<div style={div}>Cómo estás? </div> );

// Renderizamos desde la carpeta HelloWorldApp.jsx
ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp value={0} />
        
    </React.StrictMode>

);


