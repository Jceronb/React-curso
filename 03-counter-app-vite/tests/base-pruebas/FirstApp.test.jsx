import { render } from "@testing-library/react"
import { FirstApp } from "../../src/FirstApp"

describe('Pruebas en <FirstApp>', () => {

    test(' debe de hacer un match con el snapshow ', () => {

        const title = "Hola, soy Goku";    
        const {container} = render(<FirstApp title={title} /> )
        console.log(container);
        expect ( container ).toMatchSnapshot();
    });

    test('debe mostrar el tÃ­tulo en un h1 ', () => {

        const title = 'Hola, soy Goku';
        const { container, getByText } = render( <FirstApp title = {title}/> );
        expect ( getByText (title)).toBeTruthy();

        const h1 = container.querySelector('h1');
        expect (h1.innerHTML).toContain(title);
      
    });
    
    
});











