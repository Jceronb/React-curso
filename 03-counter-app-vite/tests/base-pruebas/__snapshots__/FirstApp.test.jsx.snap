// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp>  debe de hacer un match con el snapshow  1`] = `
<div>
  <h1>
    Hola, soy Goku
  </h1>
  <p>
    No hay sub título
     
  </p>
  <p>
    Jesús Cerón
  </p>
</div>
`;
